core/src/main/java/co/cask/wrangler/executor/TextDirectives.java

```diff
@@ -392,6 +394,51 @@ public TextDirectives(List<String> directives) {
         }
         break;
 
+        // parse-as-avro <column> <schema-id> <json|binary> [version]
+        case "parse-as-avro" : {
+          String column = getNextToken(tokenizer, command, "column", lineno);
+          String schemaId = getNextToken(tokenizer, command, "schema-id", lineno);
+          String type = getNextToken(tokenizer, command, "type", lineno);
+          if (!"json".equalsIgnoreCase(type) && !"binary".equalsIgnoreCase(type)) {
+           throw new DirectiveParseException(
```

Comment:
Shouldn't this type of logic be part of the directive class? Similar to config validate? Not needed here, but a future enhancement.